homeassistant/components/remote/harmony.py

```diff
@@ -184,113 +192,188 @@ def name(self):
             return
 
         device_id = None
         if device.isdigit():
-            _LOGGER.debug("Device is numeric")
+            _LOGGER.debug("%s: Device %s is numeric",
+                          self.name, device)
             if self._client.get_device_name(int(device)):
                 device_id = device
 
-        if not device_id:
-            _LOGGER.debug("Find device ID based on device name")
-            device_id = self._client.get_activity_id(str(device).strip())
+        if device_id is None:
+            _LOGGER.debug("%s: Find device ID %s based on device name",
+                          self.name, device)
+            device_id = self._client.get_device_id(str(device).strip())
 
-        if not device_id:
-            _LOGGER.error("Device  %s is invalid", device)
+        if device_id is None:
+            _LOGGER.error("%s: Device %s is invalid", self.name, device)
             return
 
         num_repeats = kwargs.get(ATTR_NUM_REPEATS)
         delay_secs = kwargs.get(ATTR_DELAY_SECS, self._delay_secs)
 
+        # Creating list of commands to send.
+        snd_cmnd_list = []
         for _ in range(num_repeats):
             for single_command in command:
-                _LOGGER.debug("Sending command %s", single_command)
-                await self._client.send_command(device, single_command)
-                await asyncio.sleep(delay_secs)
+                send_command = SendCommandDevice(
+                    device=device,
+                    command=single_command,
+                    delay=0
+                )
+                snd_cmnd_list.append(send_command)
+                if delay_secs > 0:
+                    snd_cmnd_list.append(float(delay_secs))
+
+        _LOGGER.debug("%s: Sending commands", self.name)
+        try:
+            result_list = await self._client.send_commands(snd_cmnd_list)
+        except aioexc.TimeOut:
+            _LOGGER.error("%s: Sending commands timed-out", self.name)
+            return
+
+        for result in result_list:
+            _LOGGER.error("Sending command %s to device %s failed with code "
+                          "%s: %s",
+                          result.command.command,
+                          result.command.device,
+                          result.command.code,
+                          result.command.msg
+                          )
 
     async def sync(self):
         """Sync the Harmony device with the web service."""
-        _LOGGER.debug("Syncing hub with Harmony servers")
-        await self._client.sync()
-        await self._client.get_config()
+        import aioharmony.exceptions as aioexc
+
+        _LOGGER.debug("%s: Syncing hub with Harmony cloud", self.name)
+        try:
+            await self._client.sync()
+        except aioexc.TimeOut:
+            _LOGGER.error("%s: Syncing hub with Harmony cloud timed-out",
+                          self.name)
         await self.hass.async_add_executor_job(self.write_config_file)
```

Comment:
Should we write the config even after error?

Reply:
Good point, fixed. :-)